src/vs/workbench/api/browser/mainThreadWorkspace.ts

```diff
@@ -220,4 +222,18 @@ export class MainThreadWorkspace implements MainThreadWorkspaceShape {
 	private _onDidGrantWorkspaceTrust(): void {
 		this._proxy.$onDidGrantWorkspaceTrust();
 	}
+
+	// --- edit sessions ---
+	$registerEditSessionIdentityProvider(scheme: string) {
+		this._editSessionIdentityService.registerEditSessionIdentityProvider({
+			scheme: scheme,
+			getEditSessionIdentifier: async (workspaceFolder: WorkspaceFolder, token: CancellationToken) => {
+				return this._proxy.$getEditSessionIdentifier(workspaceFolder.uri, token);
+			}
+		});
+	}
+
+	$unregisterEditSessionIdentityProvider(scheme: string) {
+		this._editSessionIdentityService.unregisterEditSessionIdentityProvider(scheme);
+	}
```

Comment:
nit, but it is nice if the service itself returns a disposable from its register-call. Otherwise anyone can come along and unregister "random" providers.

Reply:
Good point, thank you! Will create a follow up PR to avoid requiring a re-approval of this one.